# Practice Assignment

Task1:
You need to code a program that prints numbers from 1 to 20 except multiple of 2 & 3.
Hint: use "modulo" operator and "and/or" operators

In [5]:
num = int(input('enter number\n'))
for i in range(1,21):
    if i%2==0 or i%3==0:
        continue
    else:
        print(num,'*',i,'=',num*i)

enter number
3
3 * 1 = 3
3 * 5 = 15
3 * 7 = 21
3 * 11 = 33
3 * 13 = 39
3 * 17 = 51
3 * 19 = 57


Task2:
Write a program that keep on accepting number from the user until user enters Zero. Display the sum and average of all the numbers.

Hint: Use while loop with flag. 

In [7]:
nums=[]
num = int(input('enter number :'))
while num!=0:
    nums.append(num)
    num = int(input('enter number :'))
print('sum :',sum(nums))
print('average :',sum(nums)/len(nums))

enter number :3
enter number :2
enter number :1
enter number :5
enter number :0
sum : 11
average : 2.75


Task3:
Accept 10 numbers from the user and display their average.
Hint: Use loop for accepting numbers or use split to take input in one go.

In [8]:
nums = []
for i in range(10):
    nums.append(int(input('enter number: ')))
print('average :',sum(nums)/10)

enter number :2
enter number :4
enter number :2
enter number :1
enter number :6
enter number :7
enter number :8
enter number :9
enter number :12
enter number :23
average : 7.4


Task4:
Write a program to display sum of odd numbers and even numbers that fall between 12 and 37(including both numbers).

In [10]:
i=12
evens = 0
odds = 0
while i>=12 and i<=37:
    if i%2==0:
        evens+=i
    else:
        odds+=i
    i+=1
print('sum of even numbers :',evens)
print('sum of odd numbers :',odds)

sum of even numbers : 312
sum of odd numbers : 325


Task5:
Write a program to print table of a number accepted from user.

In [11]:
num = int(input('enter number: '))
for i in range(1,11):
    print(num,'*',i,'=',num*i)

enter number: 4
4 * 1 = 4
4 * 2 = 8
4 * 3 = 12
4 * 4 = 16
4 * 5 = 20
4 * 6 = 24
4 * 7 = 28
4 * 8 = 32
4 * 9 = 36
4 * 10 = 40


Task6:
Write a program to display product of the digits of a number accepted from the user.

If user inputs:123 Outputs 6

If user inputs: 567 Outputs 210

In [15]:
product = 1
num = input('enter number: ')
for i in num:
    product *= int(i)
print('product :',product)

enter number: 321
product : 6


Task7: 
Write a program to print first 10 even numbers in reverse order.

In [18]:
for i in range(20,0,-2):
    print(i)

20
18
16
14
12
10
8
6
4
2


Task8:
Write a program to find the sum of the digits of a number accepted from user

If user inputs:123 Outputs 6

If user inputs: 567 Outputs 18

In [20]:
add = 0
num = input('enter number: ')
for i in num:
    add += int(i)
print('sum :',add)

enter number: 567
sum : 18


Task9:
Write a program that accepts a sentence and calculate the number of letters and digits.

Suppose the following input is supplied to the program:

hello world! 123

Then, the output should be:

LETTERS 10

DIGITS 3

In [25]:
sentence = input('enter sentence: ')
digits = 0
letters = 0
for i in sentence:
    if i in '0123456789':
        digits+=1
    elif i==' ' or i in '<>?:!@#$%^&*':
        continue
    else:
        letters+=1
print('LETTERS',letters)
print('DIGITS',digits)

enter sentence: hello world! 123
LETTERS 10
DIGITS 3


Task10: 
You need to trace and make a list of all such numbers from1000 to 3000 in which all the digits are even numbers.

In [27]:
lst=[]
for i in range(1000, 3001,2):
    lst.append(i)
print(lst)

[1000, 1002, 1004, 1006, 1008, 1010, 1012, 1014, 1016, 1018, 1020, 1022, 1024, 1026, 1028, 1030, 1032, 1034, 1036, 1038, 1040, 1042, 1044, 1046, 1048, 1050, 1052, 1054, 1056, 1058, 1060, 1062, 1064, 1066, 1068, 1070, 1072, 1074, 1076, 1078, 1080, 1082, 1084, 1086, 1088, 1090, 1092, 1094, 1096, 1098, 1100, 1102, 1104, 1106, 1108, 1110, 1112, 1114, 1116, 1118, 1120, 1122, 1124, 1126, 1128, 1130, 1132, 1134, 1136, 1138, 1140, 1142, 1144, 1146, 1148, 1150, 1152, 1154, 1156, 1158, 1160, 1162, 1164, 1166, 1168, 1170, 1172, 1174, 1176, 1178, 1180, 1182, 1184, 1186, 1188, 1190, 1192, 1194, 1196, 1198, 1200, 1202, 1204, 1206, 1208, 1210, 1212, 1214, 1216, 1218, 1220, 1222, 1224, 1226, 1228, 1230, 1232, 1234, 1236, 1238, 1240, 1242, 1244, 1246, 1248, 1250, 1252, 1254, 1256, 1258, 1260, 1262, 1264, 1266, 1268, 1270, 1272, 1274, 1276, 1278, 1280, 1282, 1284, 1286, 1288, 1290, 1292, 1294, 1296, 1298, 1300, 1302, 1304, 1306, 1308, 1310, 1312, 1314, 1316, 1318, 1320, 1322, 1324, 1326, 1328, 1330, 133

Task11: 
You need to write a program that counts the number of vowels in a given sentence as input from console. 

Suppose the following input is supplied to the program:

Hello world! Practice makes perfect

Then, the output should be:

a appeared 2 times

e appeared 5 times

i appeared  1 time

o appeared 2 times

u appeared 0 time


In [28]:
a=0; e=0; i=0; o=0; u=0
sentence = input('enter sentence :').lower()
for p in sentence:
    if p=='a':
        a+=1
    elif p=='e':
        e+=1
    elif p=='i':
        i+=1
    elif p=='o':
        o+=1
    elif p=='u':
        u+=1
print('a appeared',a,'times')
print('e appeared',e,'times')
print('i appeared',i,'times')
print('o appeared',o,'times')
print('u appeared',u,'times')

enter sentence :Hello world! Practice makes perfect
a appeared 2 times
e appeared 5 times
i appeared 1 times
o appeared 2 times
u appeared 0 times


Task12:
You need to write a program that takes sequence of lines as input and prints the lines after making all characters in the sentence capitalized.

Suppose the following input is supplied to the program:

Hello world

Practice makes perfect

Then, the output should be:

HELLO WORLD
PRACTICE MAKES PERFECT

In [7]:
lines = ''
while True:
    l = input()
    if l!='':
        lines+=l.upper()+' '
    else:
        break;
print(lines)

fsf
bvnhgn
yujyu

FSF BVNHGN YUJYU 


Task14:
You need code a program that calculates and prints the value according to the given formula:
P = Square root of [(2 * A * B)/C]

Following are the fixed values of A and B:

A is 50. B is 30.

The values of the literal C should be taken as console input to your program in a comma-separated sequence.

Example:
Let us assume the following comma separated input sequence is given to the program:
100,150,180

The output of the program should be:
18,22,24

In [9]:
# problem
a = 50
b = 30
c = input('enter values of c: ').split(',')
for i in c:
    p = ((2*a*b)/float(i))**0.5
    print(int(p))

enter values of c: 100,150,180
5
4
4


In [10]:
##### import math
a = 50
b = 30
nuMbers = input("Please enter coMMa separated nuMbers")
if nuMbers.find(",") > -1:
    nuMbers = nuMbers.split(",")
    for i in nuMbers:
        try:
            i = int(i)
            p = math.sqrt(2 * a * b / i)
            print(p, end=",")
            
        except:
            print("Please give valid input")
else:
    print("Please give valid input")

Please enter coMMa separated nuMbers100,150,180
Please give valid input
Please give valid input
Please give valid input


Task15:
You need to write a ocde in python that finds all such numbers which are divisible by 7 but are not a multiple of 5,
between 2000 and 3200 (both included).

Print the obtained numbers in a comma-separated sequence on a single line

Hint: modulo operator, and/or operator, for loop /while loop

In [11]:
c = []
for i in range(2000,3201):
    if i%7==0 and i%5!=0:
        c.append(i)
print(c)

[2002, 2009, 2016, 2023, 2037, 2044, 2051, 2058, 2072, 2079, 2086, 2093, 2107, 2114, 2121, 2128, 2142, 2149, 2156, 2163, 2177, 2184, 2191, 2198, 2212, 2219, 2226, 2233, 2247, 2254, 2261, 2268, 2282, 2289, 2296, 2303, 2317, 2324, 2331, 2338, 2352, 2359, 2366, 2373, 2387, 2394, 2401, 2408, 2422, 2429, 2436, 2443, 2457, 2464, 2471, 2478, 2492, 2499, 2506, 2513, 2527, 2534, 2541, 2548, 2562, 2569, 2576, 2583, 2597, 2604, 2611, 2618, 2632, 2639, 2646, 2653, 2667, 2674, 2681, 2688, 2702, 2709, 2716, 2723, 2737, 2744, 2751, 2758, 2772, 2779, 2786, 2793, 2807, 2814, 2821, 2828, 2842, 2849, 2856, 2863, 2877, 2884, 2891, 2898, 2912, 2919, 2926, 2933, 2947, 2954, 2961, 2968, 2982, 2989, 2996, 3003, 3017, 3024, 3031, 3038, 3052, 3059, 3066, 3073, 3087, 3094, 3101, 3108, 3122, 3129, 3136, 3143, 3157, 3164, 3171, 3178, 3192, 3199]


In [73]:
numbers = ""
for i in range(2000, 3201):
    if i % 7 == 0 and i % 5 != 0:
        print(i, end=",")

2002,2009,2016,2023,2037,2044,2051,2058,2072,2079,2086,2093,2107,2114,2121,2128,2142,2149,2156,2163,2177,2184,2191,2198,2212,2219,2226,2233,2247,2254,2261,2268,2282,2289,2296,2303,2317,2324,2331,2338,2352,2359,2366,2373,2387,2394,2401,2408,2422,2429,2436,2443,2457,2464,2471,2478,2492,2499,2506,2513,2527,2534,2541,2548,2562,2569,2576,2583,2597,2604,2611,2618,2632,2639,2646,2653,2667,2674,2681,2688,2702,2709,2716,2723,2737,2744,2751,2758,2772,2779,2786,2793,2807,2814,2821,2828,2842,2849,2856,2863,2877,2884,2891,2898,2912,2919,2926,2933,2947,2954,2961,2968,2982,2989,2996,3003,3017,3024,3031,3038,3052,3059,3066,3073,3087,3094,3101,3108,3122,3129,3136,3143,3157,3164,3171,3178,3192,3199,

Task16:
With a given integral number n, write a program to generate a dictionary that contains (i, i*i) such that is an integral number between 1 and n (both included). and then the program should print the dictionary.

Suppose the following input is supplied to the program:

8

Then, the output should be:

{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64}

In [12]:
d={}
ask = int(input('enter n: '))
for i in range(1,ask+1):
    key = i
    val = i**2
    d[key]=val
print(d)

enter n: 7
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49}


In [76]:
squareInfo = {}
number = int(input("Enter any number"))
for i in range(1, number + 1):
    squareInfo[i] = i * i
print(squareInfo)

Enter any number8
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64}


Task17:
You need to write a code which accepts a sequence of comma separated 4 digit binary numbers as its input and then check whether they are divisible by 5 or not. 

The numbers that are divisible by 5 are to be printed in a comma separated sequence.

Example:

0100,0011,1010,1001

Then the output should be:
1010

Reference: 
1) https://www.datacamp.com/.../python-data-type-conversion

2) https://stackoverflow.com/questions/21765779/converting-binary-to-decimal-integer-output?fbclid=IwAR3DD_oiq5pn99IjyxSJgojvB5W6M4JqQtv4EUl_IF_DtmuRXlP-DswcKAA

In [82]:
number = input("Enter comma separated four digit binary number : ")
convertedNumbers = []
newStr = ""
if number.find(",") > -1:
    number = number.split(",")
    print(number)
    for i in number:
        if len(i) == 4:
            try:
                convertedNumbers.append(int(i,2))
                if convertedNumbers[len(convertedNumbers) - 1] % 5 == 0:
                    newStr = newStr + f"{i}" + ","
            except:
                print("Invalid binary number")
                break
        else:
            print("Please give valid input")
else:
    print("Please give valid input")
print(newStr)

Enter comma separated four digit binary number : 1111,1000,0101
['1111', '1000', '0101']
1111,0101,
